packages/candid/src/idl.ts

```diff
@@ -742,14 +742,20 @@ export class VecClass<T> extends ConstructType<T[]> {
   }
 
   public covariant(x: any): x is T[] {
-    return Array.isArray(x) && x.every(v => this._type.covariant(v));
+    // Special case for ArrayBuffer
+    const bits = this._type instanceof FixedNatClass ? this._type.bits : (this._type instanceof FixedIntClass ? this._type._bits : 0);
+    return (ArrayBuffer.isView(x) && bits == (x as any).BYTES_PER_ELEMENT * 8)
+           || (Array.isArray(x) && x.every(v => this._type.covariant(v)));
   }
 
   public encodeValue(x: T[]) {
     const len = lebEncode(x.length);
     if (this._blobOptimization) {
```

Comment:
Seems `_blobOptimization` is not needed anymore?

Reply:
It is still needed if people pass plain array in.

If we remove this, then old code will probably get worse performance unless they also switch from Array to Uint8Array.

Reply:
Maybe add a comment in the code. I think eventually, people want to use typedArray for large vector anyway.

Reply:
done